### ISSUE TITLE
Application crashes. Does not run on Nexus6.

### ISSUE THREAD

**Issue by AbhinavAdarsh** [HeyLookItsBrandon/VulkanTemplate#1]
04-19 19:07:29.510 13236-13236/? I/zygote64: Late-enabling -Xcheck:jni
04-19 19:07:29.614 13236-13236/com.coalminesoftware.vulkan.template W/System: ClassLoader referenced unknown path: 
04-19 19:07:29.704 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: Creating: 0x72864f9f80
04-19 19:07:29.704 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: Config: mcc=310 mnc=260 lang=en cnt=US orien=1 touch=3 dens=560 keys=1 nav=1 keysHid=3 navHid=0 sdk=27 size=2 long=1 modetype=1 modenight=1
04-19 19:07:29.706 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: Start: 0x72864f9f80
04-19 19:07:29.707 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: activityState=10
04-19 19:07:29.713 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: Resume: 0x72864f9f80
04-19 19:07:29.713 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: activityState=11
04-19 19:07:29.729 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: InputQueueCreated: 0x72864f9f80 -- 0x72864faac0
04-19 19:07:29.730 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: APP_CMD_INPUT_CHANGED
04-19 19:07:29.730 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: Attaching input queue to looper
04-19 19:07:29.764 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: NativeWindowCreated: 0x72864f9f80 -- 0x727b072010
04-19 19:07:29.765 13236-13251/com.coalminesoftware.vulkan.template V/threaded_app: APP_CMD_INIT_WINDOW
04-19 19:07:29.765 13236-13251/com.coalminesoftware.vulkan.template D/vndksupport: Loading /vendor/lib64/hw/vulkan.msm8994.so from current namespace instead of sphal namespace.
04-19 19:07:29.795 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: searching for layers in '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64'
04-19 19:07:29.795 13236-13236/com.coalminesoftware.vulkan.template V/threaded_app: WindowFocusChanged: 0x72864f9f80 -- 1
04-19 19:07:29.801 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_LUNARG_parameter_validation' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_parameter_validation.so'
04-19 19:07:29.811 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_GOOGLE_threading' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_threading.so'
04-19 19:07:29.816 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_LUNARG_object_tracker' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_object_tracker.so'
04-19 19:07:29.822 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_LUNARG_swapchain' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_swapchain.so'
04-19 19:07:29.827 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_LUNARG_image' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_image.so'
04-19 19:07:29.833 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_LUNARG_core_validation' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_core_validation.so'
04-19 19:07:29.837 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: added global layer 'VK_LAYER_GOOGLE_unique_objects' from library '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/lib/arm64/libVkLayer_unique_objects.so'
04-19 19:07:29.838 13236-13251/com.coalminesoftware.vulkan.template D/vulkan: searching for layers in '/data/app/com.coalminesoftware.vulkan.template-fICxThLTjwc4m3vmJQNctQ==/base.apk!/lib/arm64-v8a'
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: Found 9 supported instance extensions.
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_surface (25)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_android_surface (6)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_EXT_swapchain_colorspace (3)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_get_surface_capabilities2 (1)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_EXT_debug_report (8)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_get_physical_device_properties2 (1)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_external_memory_capabilities (1)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_external_semaphore_capabilities (1)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_KHR_external_fence_capabilities (1)
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: Found 7 supported validation layers.
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_LUNARG_parameter_validation (1) - LunarG Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_GOOGLE_threading (1) - Google Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_LUNARG_object_tracker (1) - LunarG Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_LUNARG_swapchain (1) - LunarG Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_LUNARG_image (1) - LunarG Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_LUNARG_core_validation (1) - LunarG Validation Layer
04-19 19:07:29.840 13236-13251/com.coalminesoftware.vulkan.template D/vulkan-template: VK_LAYER_GOOGLE_unique_objects (1) - Google Validation Layer
04-19 19:07:29.844 13236-13251/com.coalminesoftware.vulkan.template I/vulkan: Loaded layer VK_LAYER_LUNARG_parameter_validation
04-19 19:07:29.849 13236-13251/com.coalminesoftware.vulkan.template I/vulkan: Loaded layer VK_LAYER_LUNARG_core_validation
04-19 19:07:29.852 13236-13251/com.coalminesoftware.vulkan.template I/Adreno: QUALCOMM build          : 8e59954, I0be83d0d26
                                                                              Build Date              : 09/22/17
                                                                              Shader Compiler Version : EV031.21.02.02
                                                                              Local Branch            : O17A
                                                                              Remote Branch           : 
                                                                              Remote Branch           : 
                                                                              Reconstruct Branch      : 
04-19 19:07:29.854 13236-13251/com.coalminesoftware.vulkan.template A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x1cdc0de in tid 13251 (vulkan.template), pid 13236 (vulkan.template)

**Comment by HeyLookItsBrandon** [comment]
Hi @AbhinavAdarsh! I experience the same crash on my nVidia Shield TV. There appears to be an issue with some of the validation layers provided in the Android Vulkan SDK.

I have had luck getting some (but not all) of them running by recompiling them from source. You can also fix the crash by commenting out `VK_LAYER_LUNARG_core_validation` from the list of layers. This is unfortunate because that is one of the most important useful validation layers.

With that said, know that this project is still a work in progress and does not render anything yet.

**Comment by AbhinavAdarsh** [comment]
Thanks @HeyLookItsBrandon. I was just looking to create a simple compute example, for example to add two vectors or just copy a vector to another. I looked into headless compute example too by https://github.com/SaschaWillems/Vulkan. If you have any resource which would be a good starting point for creating a simple compute example for android without any surface, kindly point it out.

**Comment by HeyLookItsBrandon** [comment]
@AbhinavAdarsh Sorry, I wish I had an example but I don’t. I’ve been working through the lesson from vulkan-tutorial.com and adapting it for Android. It’s specific to graphics though. Vulkan is new enough, that you might have a hard time finding compute examples for Android unless there is one buried somewhere in the Google examples at https://github.com/googlesamples/vulkan-basic-samples

I’m not sure how much flexibility you have for picking a library but at this point, something like RenderScript or OpenCL might be better documented on Android.

**Comment by HeyLookItsBrandon** [comment]
@AbhinavAdarsh I was thinking about this again and realized that I was providing an invalid address in the `…CreateInfo` instances. In VulkanNativeApp.cpp, I define a `std::vector` of validation layer names but rather than using it directly, I was calling a method I wrote to query the available layers and filter out the ones that aren't available. I was initializing the `…CreateInfo` instances to that new `vector`'s size and data but the vector goes out of scope before the `vkCreate…()` call, so by the time that happens, the address is no longer valid and the behavior is undefined.

I fixed that in a recent commit. It sounded like you were looking for a different type of project than this anyway but I figured you might be interested if you were referencing this from time to time. Did you find good, concise a compute example for Android?
